src/js/tech/tech.js

```diff
@@ -765,19 +765,27 @@ class Tech extends Component {
   }
 
   /**
-   * Create a floating video window always on top of other windows so that users may
-   * continue consuming media while they interact with other content sites, or
-   * applications on their device.
+   * Attempt to create a floating video window always on top of other windows
+   * so that users may continue consuming media while they interact with other
+   * content sites, or applications on their device.
    *
    * @see [Spec]{@link https://wicg.github.io/picture-in-picture}
    *
-   * @return {Promise}
-   *         A promise with a Picture-in-Picture window.
+   * @return {Promise|undefined}
+   *         A promise with a Picture-in-Picture window if the browser supports
+   *         Promises (or one was passed in as an option). It returns undefined
+   *         otherwise.
    *
    * @abstract
    */
   requestPictureInPicture() {
-    return Promise.reject();
+    const PromiseClass = this.options_.Promise || window.Promise;
+
+    if (PromiseClass) {
+      return new PromiseClass((resolve, reject) => {
```

Comment:
I think this can just be `return PromiseClass.reject()`, the polyfill is expected to have the static methods too

Reply:
Fixed in https://github.com/videojs/video.js/pull/6001/commits/4cbe4139d2a54c5182ea653b0e25185720b15cae